torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -589,6 +590,16 @@ ProcessGroupNCCL::ProcessGroupNCCL(
             << "\nUSE_HIGH_PRIORITY_STREAM: "
             << options_->is_high_priority_stream
             << "\nNCCL_DEBUG: " << ncclDebugLevel;
+
+#ifdef USE_NCCL_WITH_UCC
+  static std::once_flag initialize_ucc_lib_flag;
+  std::call_once(initialize_ucc_lib_flag, [&]{
```

Comment:
Why do we need to pass `&` into the lambda? `uccLib_` is static and we shouldn't need `&` to access it in the lambda?

Reply:
This is for `rank_`, otherwise we will have
```
/home/gaoxiang/pytorch-ucc2/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp: In lambda function:
/home/gaoxiang/pytorch-ucc2/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:599:32: error: ‘this’ was not captured for this lambda function
  599 |       LOG(INFO) << "[Rank " << rank_  << "] torch_ucc.so loaded";
      |                                ^~~~~
/home/gaoxiang/pytorch-ucc2/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:599:32: error: invalid use of non-static data member ‘c10d::ProcessGroup::rank_’
In file included from /home/gaoxiang/pytorch-ucc2/torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp:13,
                 from /home/gaoxiang/pytorch-ucc2/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:1:
/home/gaoxiang/pytorch-ucc2/torch/csrc/distributed/c10d/ProcessGroup.hpp:431:13: note: declared here
  431 |   const int rank_;
      |             ^~~~~

```